src/middleware.ts

```diff
@@ -137,6 +146,36 @@ export async function middleware(req: NextRequest, ev: NextFetchEvent) {
 		response = NextResponse.rewrite(url)
 	}
 
+	// Check for the `variants` cookie
+	//  * Check if the current tutorial path is on the variant rewrites sheet
+	//  * If so, check for the variant object
+	//  * If present, redirect to the option
+
+	// if the pathname includes tutorials,
+	// the path is included in variants data as having a variant
+	// the cookie exists and has the associated value
+
+	if (req.cookies.get('variants') && variantRewrites[req.nextUrl.pathname]) {
+		const url = req.nextUrl.clone()
+		const tutorialVariant = variantRewrites[req.nextUrl.pathname]
+		let variantOption
+
+		try {
+			const cookie = req.cookies.get('variants')
+			const allVariantsCookie = JSON.parse(cookie.value)
+			variantOption = allVariantsCookie[tutorialVariant.slug]
+		} catch (e) {
+			console.log(e)
+		}
+
+		// rewrite to the static route
+		url.pathname = `${url.pathname}/${getVariantParam(
+			tutorialVariant.slug,
+			variantOption
+		)}`
+		response = NextResponse.rewrite(url)
+	}
```

Comment:
🧼 Will clean this up and see if we can consolidate with the other variants logic well.